lib/ai/aiCoder.ts

```diff
@@ -62,6 +65,7 @@ export class AiCoderImpl implements AiCoder {
 export const createAiCoder = (
   onStreamedChunk: (chunk: string) => void,
   onVfsChanged: () => void,
+  anthropicClient?: import("@anthropic-ai/sdk").Anthropic,
 ): AiCoder => {
-  return new AiCoderImpl(onStreamedChunk, onVfsChanged)
+  return new AiCoderImpl(onStreamedChunk, onVfsChanged, anthropicClient)
```

Comment:
should be broken into named parameters https://github.com/tscircuit/handbook/blob/main/guides/code.md#1-two-parameter-rule

Reply:
also we can make `AiCoderImpl` extend `EventEmitter` or have a `.on("...")` function to be a bit more conventional, didn't realize this until i saw it now